tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +761,253 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+			srcClusterStatus, srcClusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(srcClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying source cluster %s status", SourceClusterName))
+			destClusterStatus, _ = Inst().Backup.RegisterBackupCluster(orgID, destinationClusterName, "")
+			dash.VerifyFatal(destClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying destination cluster %s status", destinationClusterName))
+		})
+		Step("Getting the replica factor of mongodb statefulset in backup namespace before taking backup", func() {
+			log.InfoD("Getting the replica factor of mongodb statefulset in backup namespace before taking backup")
+			backupNS, err = backup.GetPxBackupNamespace()
+			dash.VerifyFatal(err, nil, "Getting backup namespace")
+			statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, backupNS)
+			originalReplicaCount = *statefulSet.Spec.Replicas
+			log.Infof("Number of replica for mongodb pod before backup is %v", originalReplicaCount)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Getting mongodb statefulset replica in backup namespace %s", backupNS))
+			dash.VerifyFatal(originalReplicaCount > scaledDownReplica, true, "Verifying mongodb statefulset replica before taking backup")
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting admin context")
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					backupName := fmt.Sprintf("%s-%s-%d", BackupNamePrefix, namespace, i)
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateBackupWithoutCheck(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+							[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of backups taken are: %v", backupNames)
+		})
+		Step("Scaling MongoDB statefulset replica to 0 and back to original replica while backup is in progress", func() {
+			log.InfoD("Scaling MongoDB statefulset replica to 0 while backup is in progress")
+			*statefulSet.Spec.Replicas = scaledDownReplica
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling to 0 is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == scaledDownReplica, true, "Verify mongodb statefulset replica after scaling down")
```

Comment:
Wouldn't it be better if we retrieve the spec replicas using GetStatefulSet and compare it instead?

Reply:
I apologize for my oversight. I did not notice that UpdateStatefulSet is also returning a statefulset.